### ISSUE TITLE
Advisor tab resource ID instead of name displayed

### ISSUE THREAD

**Issue by Stel-TG** [microsoft/ARI#217]
When running the script either in the AzureCLI or via Automation account the name column in the Advisor tab contains mostly the resource ID's instead of the resource name.  The names appear correctly on the tabs and if I execute the PowerShell script from my desktop using the same account logged into the AzurePortal.

**Comment by Claudio-Merola** [comment]
Hello @Stel-TG 

I just published the version 3.5.1 of the module. Can you try to upgrade your version of the module and try again?

Thanks

**Comment by Stel-TG** [comment]
I tried both the 3.51 and 3.52 updates and several resources show the name as the subscription ID in both versions. The other thing that changed is the same behavior exists when I run it locally now

**Comment by Stel-TG** [comment]
It should be noted that some item names do show up in the advisor tab.  I tried correlating to find commonalities around the impact/type and while most that showed up as the subscriptionID for the resource name we cost and high availability related.  I have several resource types that arent resolving but things like virtual machines some recommendations show the name while others show the subscriptionID.  The only difference is the resource type Microsoft.Compute/virtualMachines and all the resource types whose names resolve properly the resource types are completely upper case. ie MICROSOFT.COMPUTE/VIRTUALMACHINES

**Comment by Claudio-Merola** [comment]
@Stel-TG 

I'm having a hard time to understand what exactly might be the issue internally in the script side.

Can you post a screenshot of the issue? You can change the resource names and ids, just to make it easier for me to better understand the issue

**Comment by Stel-TG** [comment]
@Claudio-Merola 

Yes I can see how it's confusing.  I've done this on different subscriptions with the same results


![image](https://github.com/user-attachments/assets/765e886a-a073-4fb1-88f5-884c820b1360)

**Comment by Claudio-Merola** [comment]
Hi @Stel-TG 

Thanks for the screenshot. 

This is actually right. The thing is that advisor recommendations are not always target to resources, indeed most of the recommendations are target to the entire subscription. 

So for the recommendations that do have the target resources we present it, but when the recommendations only have the information about the subscription, we can only show the subscription ID
